pkg/controller/extensions/request/extensionrequest.go

```diff
@@ -515,18 +519,20 @@ func (r *Reconciler) discoverExecutorInfo(ctx context.Context) error {
 
 // discoverExecutorInfo is the concrete implementation that will lookup executorInfo from the runtime environment.
 func (d *executorInfoDiscoverer) discoverExecutorInfo(ctx context.Context) (*executorInfo, error) {
-	pod, err := util.GetRunningPod(ctx, d.kube)
-	if err != nil {
-		log.Error(err, "failed to get running pod")
-		return nil, err
-	}
+	image := os.Getenv(PodImageNameEnvVar)
```

Comment:
one way to avoid these indentation changes is to invert this condition.  if we got a non-empty string, we could immediately return it.  e.g.,
```
if image != "" {
  return &executorInfo{image: image}, nil
}

// rest of code remains the same i think
```

Reply:
I was trying to avoid repeating the main return statement.  Consider this flattened (473bb0a0f6134ba43ff3210c6c0f341f809bc8d0).